src/app-sync.test.js

```diff
@@ -75,3 +75,65 @@ describe('/download-user-file', () => {
     });
   });
 });
+
+describe('/delete-user-file', () => {
+  it('returns 401 if the user is not authenticated', async () => {
+    const res = await request(app).post('/delete-user-file');
+
+    expect(res.statusCode).toEqual(401);
+    expect(res.body).toEqual({
+      details: 'token-not-found',
+      reason: 'unauthorized',
+      status: 'error',
+    });
+  });
+
+  // it returns 422 if the fileId is not provided
+  it('returns 422 if the fileId is not provided', async () => {
+    const res = await request(app)
+      .post('/delete-user-file')
+      .set('x-actual-token', 'valid-token');
+
+    expect(res.statusCode).toEqual(422);
+    expect(res.body).toEqual({
+      details: 'fileId-required',
+      reason: 'unprocessable-entity',
+      status: 'error',
+    });
+  });
+
+  it('returns 400 if the file does not exist', async () => {
+    const res = await request(app)
+      .post('/delete-user-file')
+      .set('x-actual-token', 'valid-token')
+      .send({ fileId: 'non-existing-file-id' });
+
+    expect(res.statusCode).toEqual(400);
+    expect(res.text).toEqual('file-not-found');
+  });
+
+  it('marks the file as deleted', async () => {
+    const accountDb = getAccountDb();
+    const fileId = 'existing-file-id';
+
+    // Insert a file into the database
+    accountDb.mutate(
+      'INSERT OR IGNORE INTO files (id, deleted) VALUES (?, FALSE)',
+      [fileId],
+    );
+
+    const res = await request(app)
+      .post('/delete-user-file')
+      .set('x-actual-token', 'valid-token')
+      .send({ fileId });
+
+    expect(res.statusCode).toEqual(200);
+    expect(res.body).toEqual({ status: 'ok' });
+
+    // Verify that the file is marked as deleted
+    const rows = accountDb.all('SELECT deleted FROM files WHERE id = ?', [
+      fileId,
+    ]);
+    expect(rows[0].deleted).toBe(1);
```

Comment:
tiny nit, I think this would read more obviously as `true` if possible

```suggestion
    expect(rows[0].deleted).toBe(true);
```

Reply:
I had it add `true` at the beginning, but it's actually `1`, not `true`.
I could always add `const true = 1` and use the variable, but that seems kinda wrong. 
It might be a good idea in the future to wrap the database return objects in a custom class, so that this could be typed + transformed into a boolean.

Reply:
Ah of course, it's an integer field, sorry!

Reply:
We could use `.toBeTruthy()`?

https://jestjs.io/docs/expect#tobetruthy

Reply:
Proof:

```
 FAIL  src/app-sync.test.js
  ● /delete-user-file › marks the file as deleted

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: 1

      135 |       fileId,
      136 |     ]);
    > 137 |     expect(rows[0].deleted).toBe(true);
          |                             ^
      138 |   });
      139 | });
      140 |

      at Object.toBe (src/app-sync.test.js:137:29)
```